### PR TITLE
Code coverage improvement for set algorithms

### PR DESCRIPTION
**Pull request by dparanic**
set_union, set_intersection and other set algorithms are now also tested without comparator.

### CODE REVIEW

File: test/parallel_api/algorithm/alg.sorting/alg.set.operations/set.pass.cpp [hunk 1]
```diff
@@ -281,14 +381,14 @@ main()
         int32_t;
 #endif
 
-    test_set<data_t, data_t>(oneapi::dpl::__internal::__pstl_less());
+    test_set<data_t, data_t>(oneapi::dpl::__internal::__pstl_less(),false);
```
Comment (capatober) [inline test/parallel_api/algorithm/alg.sorting/alg.set.operations/set.pass.cpp:384]: Tests below are not testing when `TEST_DPCPP_BACKEND_PRESENT==1`. Could you please add a test with `true`?

Comment (dparanic) [reply test/parallel_api/algorithm/alg.sorting/alg.set.operations/set.pass.cpp:384]: Done

Comment (dparanic) [reply test/parallel_api/algorithm/alg.sorting/alg.set.operations/set.pass.cpp:378]: Added

**Review by capatober — APPROVED** [review]

**Comment by dparanic** [comment]
CI on RHEL has failed due to problem with machine. PR is merged because CI on other configurations has completed successfully.

**Pull request merged**